web-api/src/storage/elastic/filter.rs

```diff
@@ -67,6 +68,35 @@ impl Serialize for Terms<'_> {
     }
 }
 
+#[derive(Debug)]
+struct Parents<'a> {
+    parents: &'a [DocumentId],
+}
```

Comment:
This is a bit early to include into this PR but when excluding document/snippets we need:

- exclude all snippets for a document => this needs `must_not: [ { terms: { parent: [values...] } } ]`
- exclude all old documents which don't have a parent field by id => `must_not: [ { ids: [values...] } ]`
- exclude all snippets where only the snippet is excluded => `must_not: [ { ids: [es_snippet_ids...] } ]`